fhirclient/client.py

```diff
@@ -240,3 +243,97 @@ def from_state(self, state):
     def save_state (self):
         self._save_func(self.state)
 
+    # MARK: Pagination
+    def fetch_next_page(self, bundle):
+        """
+        Fetch the next page of results using the `next` link provided in the bundle.
+
+        Args:
+            bundle (Bundle): The FHIR Bundle containing the `next` link.
+
+        Returns:
+            Bundle: The next page of results as a FHIR Bundle.
+        """
+        next_link = self.get_next_link(bundle)
+        if next_link:
+            sanitized_next_link = self.sanitize_next_link(next_link)
+            return self.execute_pagination_request(sanitized_next_link)
+        return None
```

Comment:
This isn't a bad approach, but feels like it's so close to being a really Pythonic iter() pattern. I don't have a strong preference, but curious if you considered that approach. Not sure what the API would look like for that. Maybe something like?

```
for bundle in client.iter_pages(first_bundle):
````

Not sure I love that either, but the current use wouldn't be too far off:
```
bundle = first_bundle
while bundle := client.fetch_next_page(bundle):
````

Reply:
There are also a lot of ways to do the iteration approach. Off the top of my head:

- We could add __iter__ and __next__ fields to the Bundle object directly...
- We could add FHIRSearch.perform_iter() as a generator maybe that yields Bundle objects...
- Or the above FHIRClient.iter_pages() approach.

Reply:
Thank you very much for all your ideas and comments. The latest changes are pushed. 
Since backward compatibility is a concern, I kept all changes in FHIRClient class. If existing code relies on Bundle objects being iterated over using traditional loops or other methods that expect a list-like behavior, adding __iter__ and __next__ might interfere with these expectations. Users might also have existing methods that do not anticipate Bundle objects to be iterable in this specific way. Please let me know if you disagree and want me to change code further.

Reply:
That sounds like a great plan. I agree that embedding pagination functionality within the Bundle class could streamline the API and make it more intuitive. It seems like this might be a good candidate for a separate ticket to ensure it gets the attention it deserves. I’d be happy to create the ticket and contribute to the changes under your guidance if that works for you.

Reply:
FYI: I did summarize the 4.x discussion in a ticket, #172